src/json2typescript/json-convert.ts

```diff
@@ -661,9 +665,18 @@ export class JsonConvert {
         }
 
 
-        // Check if the property is optional
-        // If the json value is null, we don't assign it in that case
-        if (isOptional && classInstancePropertyValue === null) return;
+        // Check if the json value is null.
+        // If the property is optional, then drop the property.
+        // If the property is not optional, assign null value to json.
+        if (classInstancePropertyValue === null) {
+            if (isOptional === PropertyConvertingMode.ALWAYS_OPTIONAL ||
+                isOptional === PropertyConvertingMode.SERIALIZE_OPTIONAL) {
+                return;
+            } else {
+                json[jsonPropertyName] = null;
+                return;
+            }
+        }
```

Comment:
The `else` code in the serialization is probably the critical part, because it changes the behavior a little bit from now:

I set the json property value to `null` no matter what type the TypeScript class defines. I simply ignore if it was a primitive type, object, array or whatever. Earlier behavior was an Error, so it is not a breaking change.

Thoughts: If I allow a TypeScript property to be null even though the mapper `String` was chosen, it must be a unified type at property level (`string | null`). If I am in the else-clause, the property is NOT optional for serialization. Thus, I allow null to be set in the json.

Not sure if that makes sense lol.

Reply:
Definitely makes sense!  I like that this helps to enforce that typing.  I'll definitely be using these new options.